components/brave_rewards/resources/extension/brave_rewards/background/storage.ts

```diff
@@ -47,7 +47,8 @@ const cleanData = (state: RewardsExtension.State) => {
   state = { ...state }
   state.publishers = {}
 
-  if (!state.balance) {
+  const balance = state.balance as any
```

Comment:
we need to convert it to any, so that TS definition doesn't return lint error. Lint error is caused because definition doesn't allow you to have undefined, but in upgrade path this is the case. We handle upgrade case here.